### DISCUSSION TITLE
Suggestion: replace Xinput with SDL

### DISCUSSION THREAD

**Discussion by joao678** [samuelgr/Xidi#55] [Ideas]
Why make the switch? Well, let me introduce you to a few problems:

1. First and most important issue currently with xidi is: it does not support Playstation or (I assume) Nintendo Switch controllers
2. To use a Playstation controller you'd have to use some kind of wrapper to xinput like DS4Windows

But why SDL? Let me list a few advantages over xinput:

1. It supports pretty much every controller not only Playstation or switch controllers, but also xbox controllers so nothing is lost
2. It eliminates the need for external software to "wrap" controllers to xinput
3. It has a unified mapping system so every controller is read the same eg. "X" on Playstation is "A" in SDL, "A" on Xbox is "A" on SDL so nothing is lost
4. Not only it has vibration support but also LED player indicators and LED rgb colors for playstation controllers, trigger vibration, adaptive trigger support for DualSense controllers (this can be useful to emulate physical buttons on triggers on games that used buttons instead of axis in R2 and L2 and gyroscope/accelerometer support (maybe this can be used in older racing games to map the gyroscope into a direcinput axis?)

so basically to my limited knowlege xidi works like this: take xinput controller buttons and axis -> translate them into legacy directinput buttons and axis
My suggestion is: take SDL controller buttons and axis -> translate them into legacy directinput buttons and axis
